### PR TITLE
RX TX queue tuning before running iperf test

### PR DESCRIPTION
**Pull request by vaishnavibhat**
The patch enables runnign of irqbalance service, tuning rx tx queue values before starting iperf tests.

The command to run iperf test from client side is improvised to include parallel threads, interval between periodic bandwidth reports and time in seconds to transmit for values

### CODE REVIEW

**Comment by PraveenPenguin** [comment]
@vaishnavibhat can you please rebase as we have issue in Travis CI which was address in other commit

File: io/net/iperf_test.py [hunk 1]
```diff
@@ -183,29 +184,61 @@ def nping(self):
...
+
+    def tune_rxtx_queue(self):
+        """
+        Increase rx/tx queues to 16 for test interface
+        """
+        cmd = "ethtool -L %s rx 16 tx 16" % self.peer_interface
+        output = self.session.cmd(cmd)
+        if not output:
+            self.cancel("Unable to tune RX and TX queue in peer")
+        cmd = "ethtool -L %s rx 16 tx 16" % self.iface
```
Comment (abdhaleegit) [inline io/net/iperf_test.py:211]: rx tx number 16 might not work for all types of devices like dedicated, vnic, sriov  ?

File: io/net/iperf_test.py [hunk 2]
```diff
@@ -183,29 +184,61 @@ def nping(self):
...
+        for line in process.system_output(cmd_status).decode("utf-8").splitlines():
+            if re.search("Active: active (running)", line):
+                self.log("irqbalance service is active in host")
+
+    def tune_rxtx_queue(self):
+        """
+        Increase rx/tx queues to 16 for test interface
+        """
+        cmd = "ethtool -L %s rx 16 tx 16" % self.peer_interface
+        output = self.session.cmd(cmd)
```
Comment (abdhaleegit) [inline io/net/iperf_test.py:208]: only change values for peer interface.. and not host_interface ?

**Review by abdhaleegit — CHANGES_REQUESTED** [review]
@vaishnavibhat 2 questions
1. are these changes applicable for all type of devices or only vnic
2. should other performance test requires similar change... in that case move some code can be moved to common like enable irq and tune

File: io/net/iperf_test.py (hunk 1 above)
Comment (vaishnavibhat) [reply io/net/iperf_test.py:211]: ethtool test which tests for different rx, tx queue size works for all kind of interfaces. So this change should not be a problem.

Comment (vaishnavibhat) [reply io/net/iperf_test.py:211]: cmd = "ethtool -L %s rx 16 tx 16" % self.iface 
      This part is taking care of changing the queue value for host interface

File: io/net/iperf_test.py (hunk 2 above)
Comment (vaishnavibhat) [reply io/net/iperf_test.py:208]: cmd = "ethtool -L %s rx 16 tx 16" % self.iface
This part is taking care of changing the queue value for host interface

**Comment by vaishnavibhat** [comment]
> @vaishnavibhat 2 questions
> 
> 1. are these changes applicable for all type of devices or only vnic
> 2. should other performance test requires similar change... in that case move some code can be moved to common like enable irq and tune

Yes, sending patch to include the change in netperf and uperf test cases

**Comment by abdhaleegit** [comment]
1. move ethtool rx tx settings to individual test
2. and perf specific values settings to iperf and netperf test
3. common utils like irqbalance to utils

**Comment by vaishnavibhat** [comment]
avocado run iperf_test.py -m iperf_virt_test.yaml --max-parallel-tasks=1 
JOB ID     : 62add810eb852020230ddc2d3dc7ae96a8d52d6c
JOB LOG    : /root/avocado-fvt-wrapper/results/job-2024-04-03T00.12-62add81/job.log
 (1/2) iperf_test.py:Iperf.test;run-mtu-1500-8a94: STARTED
 (1/2) iperf_test.py:Iperf.test;run-mtu-1500-8a94: PASS (53.51 s)
 (2/2) iperf_test.py:Iperf.test;run-mtu-9000-92c4: STARTED
 (2/2) iperf_test.py:Iperf.test;run-mtu-9000-92c4: FAIL: FAIL: Throughput Actual - 68.0%, Expected - 80%, Throughput Actual value - 17000Mb/sec  (64.77 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /root/avocado-fvt-wrapper/results/job-2024-04-03T00.12-62add81/results.html
JOB TIME   : 137.05 s

Test summary:
2-iperf_test.py:Iperf.test;run-mtu-9000-92c4: FAIL

[debug.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/14842479/debug.log)

[debug.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/14842490/debug.log)

[job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/14842516/job.log)

**Comment by abdhaleegit** [comment]
@vaishnavibhat an existing utils for is_vnic https://github.com/avocado-framework/avocado/blob/76199b2472363dc5963c7dc8636ca47c2250a4db/avocado/utils/network/interfaces.py#L633

**Comment by abdhaleegit** [comment]
@vaishnavibhat anyting pending on this ? I see the existing is_vnic utils unused

**Comment by vaishnavibhat** [comment]
> @vaishnavibhat anyting pending on this ? I see the existing is_vnic utils unused

Taken care of the changes to include the utils is_vnic()

**Review by abdhaleegit — APPROVED** [review]
LGTM

**Comment by PraveenPenguin** [comment]
@vaishnavibhat please rebase PR , and address CI failure

**Review by abdhaleegit — APPROVED** [review]
Thanks for the PR

**Pull request merged**